File: sdk/insight-stream.js
// insight-stream Node.js SDK (basic)
const axios = require('axios');

class InsightStream {
  constructor({ baseUrl, apiKey }) {
    this.baseUrl = baseUrl;
    this.apiKey = apiKey;
  }

  async ingestLog(labels, message, timestamp = new Date().toISOString()) {
    const payload = {
      streams: [
        {
          labels,
          entries: [{ ts: timestamp, line: message }],
        },
      ],
    };
    const headers = { 'Content-Type': 'application/json' };
    if (this.apiKey) headers['X-API-Key'] = this.apiKey;
    const res = await axios.post(`${this.baseUrl}/ingest`, payload, { headers });
    return res.data;
  }

  async queryLogs(query, limit = 100) {
    const headers = {};
    if (this.apiKey) headers['X-API-Key'] = this.apiKey;
    const res = await axios.get(`${this.baseUrl}/query`, {
      params: { query, limit },
      headers,
    });
    return res.data;
  }
}

module.exports = InsightStream;

File: sdk/insight_stream.py
# insight_stream Python SDK (basic)
import requests
from typing import Dict, Optional

class InsightStream:
    def __init__(self, base_url: str, api_key: Optional[str] = None):
        self.base_url = base_url
        self.api_key = api_key

    def ingest_log(self, labels: Dict[str, str], message: str, timestamp: Optional[str] = None):
        import datetime
        if not timestamp:
            timestamp = datetime.datetime.utcnow().isoformat() + 'Z'
        payload = {
            "streams": [
                {
                    "labels": labels,
                    "entries": [{"ts": timestamp, "line": message}]
                }
            ]
        }
        headers = {"Content-Type": "application/json"}
        if self.api_key:
            headers["X-API-Key"] = self.api_key
        resp = requests.post(f"{self.base_url}/ingest", json=payload, headers=headers)
        resp.raise_for_status()
        return resp.json()

    def query_logs(self, query: str, limit: int = 100):
        headers = {}
        if self.api_key:
            headers["X-API-Key"] = self.api_key
        resp = requests.get(f"{self.base_url}/query", params={"query": query, "limit": limit}, headers=headers)
        resp.raise_for_status()
        return resp.json()
